targets/alert-cli/Dockerfile

```diff
@@ -1,36 +1,48 @@
 ARG NODE_VERSION=20.3.1-alpine
 
-# dist
-FROM node:$NODE_VERSION AS dist
+FROM node:$NODE_VERSION AS deps
 
-WORKDIR /
-
-COPY . /
+WORKDIR /app
 
-RUN yarn --frozen-lockfile
+COPY yarn.lock .yarnrc.yml ./
+COPY .yarn .yarn
+RUN yarn fetch workspaces focus alert-cli
 
+FROM deps AS build-graphql-client
+COPY shared/graphql-client ./shared/graphql-client
 RUN yarn workspace @shared/graphql-client build
 
+
+FROM deps AS build-dila-resolver
+COPY shared/types ./shared/types
+COPY shared/dila-resolver ./shared/dila-resolver
 RUN yarn workspace @shared/dila-resolver build
 
+FROM deps AS dist
+
+COPY --from=build-graphql-client /app/shared/graphql-client /app/shared/graphql-client
+COPY --from=build-dila-resolver /app/shared/dila-resolver /app/shared/dila-resolver
+
+COPY shared/types ./shared/types
+COPY targets/alert-cli ./targets/alert-cli
+COPY tsconfig.build.json ./
+
 RUN yarn workspace alert-cli build
 
-# app
+RUN yarn workspaces focus --production alert-cli && yarn cache clean
+
+
 FROM node:$NODE_VERSION
 
 # Add git to docker image
 RUN apk add --no-cache git openssh-client
 
-WORKDIR /app
+USER 1000
 
-USER node
+ENV NODE_ENV=production
 
-COPY --from=dist targets/alert-cli/package.json /app/alert-cli/package.json
-COPY --from=dist targets/alert-cli/dist/ /app/alert-cli/dist/
-COPY --from=dist targets/alert-cli/node_modules/ /app/alert-cli/node_modules/
-COPY --from=dist package.json /app/package.json
-COPY --from=dist node_modules /app/node_modules
+WORKDIR /app
 
-ENV NODE_ENV=production
+COPY --from=dist /app/targets/alert-cli/dist ./dist/
 
-CMD ["yarn", "--cwd", "alert-cli", "start"]
+CMD ["node","--enable-source-maps","--unhandled-rejections=strict","/app/dist/index.js"]
```

Comment:
c'est une best practice de ne pas utiliser npm ou yarn qui n'est pas un process manager et ne va pas relayer les signaux de shutdown correctement à node, voir:
- https://snyk.io/fr/blog/10-best-practices-to-containerize-nodejs-web-applications-with-docker/
- https://medium.com/@nodepractices/docker-best-practices-with-node-js-e044b78d8f67
```
use CMD ['node','server.js'] to start your app, avoid using npm scripts which don't pass OS signals to the code. This prevents problems with child-process, signal handling, graceful shutdown and having processes.

🚩 Otherwise: When no signals are passed, your code will never be notified about shutdowns. Without this, it will lose its chance to close properly possibly losing current requests and/or data.
```

mais je viens de voir que yarn a discrètement résolu le problème sur berry en 2021(je viens d'upgrade ma connaissance): https://github.com/yarnpkg/berry/pull/2347
good news !!! donc je vais faire le change et réutiliser yarn ;-)

Reply:
fixed